### PR TITLE
Fix hiding navigation on root folder

### LINKED ISSUE #19493: File-picker doesn't show folders when "Root" folder is set
### Describe the Bug

When there's no root folder, the file picker drawer has the file library navigation included:

CleanShot 2023-08-22 at 17 40 18@2x

This is missing when you're using a root folder:

CleanShot 2023-08-22 at 17 41 30@2x

Even when you have a root folder, you should be able to use the file sidebar navigation

### To Reproduce

Use the file interface with the root folder option configured, select an existing file from the system

### Directus Version

10.5.3

### Hosting Strategy

Directus Cloud

### PR DESCRIPTION
**Pull request by Nitwel**
Fixes #19493 

Screenshot 2023-09-15 at 03-36-43 Directus

### CODE REVIEW

**Review by paescuj — DISMISSED** [review]
We need to introduce new options to address this - simply showing the navigation in every case would be a breaking change, see https://github.com/directus/directus/issues/19493#issuecomment-1691575197.

**Comment by paescuj** [comment]
Closing for now in favor of new pull request which addresses https://github.com/directus/directus/pull/19518#issuecomment-1691656913.

**Comment by Nitwel** [comment]
Added the option now to show or hide the drawer manually. And when the root folder is set, it properly only shows that folder and it's children.

Also added tests for the useFiles composable. 🧪 

grafik

grafik

**Review by DanielBiegler — DISMISSED** [review]
There seems to be a problem with refetching the folders after creating/deleting/moving folders

[Screencast from 04.09.2023 15:37:27.webm](https://github.com/directus/directus/assets/14810858/b793fa24-a5a5-4650-9171-aa668542d71a)

Switched back to main branch for comparison and dont have the problem there.

**Review by paescuj — APPROVED** [review]

File: app/src/views/private/components/files-navigation.vue [hunk 1]
```diff
@@ -84,12 +86,18 @@ const props = defineProps<{
...
-const { nestedFolders, folders, loading, openFolders: sharedOpenFolders } = useFolders();
-const openFolders = props.localOpenFolders ? ref(openFoldersInitial) : sharedOpenFolders;
+const { nestedFolders, folders, loading, openFolders } = useFolders(rootFolder, localOpenFolders);
 
 watch([() => props.currentFolder, loading], setOpenFolders, { immediate: true });
 
+const rootFolderInfo = computed(() => {
+	if (!folders.value || !props.rootFolder) return;
+
+	return folders.value.find((folder) => folder.id === rootFolder!.value);
```
[the remark this answers is not in the document]
Comment (paescuj) [reply app/src/views/private/components/files-navigation.vue]: Great catch ❤️

**Review by azrikahar — APPROVED** [review]

**Pull request merged**